src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -277,10 +273,8 @@ uint G1Policy::calculate_young_desired_length(size_t pending_cards, size_t rs_le
 
     // Incorporate MMU concerns; assume that it overrides the pause time
     // goal, as the default value has been chosen to effectively disable it.
-    // Also request at least one eden region, see above for reasons.
-    uint desired_eden_length = MAX3(desired_eden_length_by_pause,
-                                    desired_eden_length_by_mmu,
-                                    MinDesiredEdenLength);
```

Comment:
Does this mean `G1Policy::calculate_young_desired_length()` could return 0 instead of a minimum of 1 now? Comment on line 228 may need updating.

I noticed `young_list_desired_length()` is also used in `G1Policy::update_ihop_prediction()`. Could this change mess up with `_ihop_control->update_allocation_info()`?

Reply:
No. Line 285 clamps the value to min/max of the young gen sizer; minimum size returned by it is 1 region, see `G1YoungSizer` code. So the original code to ensure a minimum region size of one has already been superfluous in the original implementation.
Since it serves no purpose, I removed it. I added an assertion at the end.

Reply:
> Thank you for the quick fix!
> 
> I haven't fully understood JDK-8253413 yet, so better to have someone more familiar with JDK-8253413 to review this change.

Feel free to ask if there is something you do not understand.